runner/cf-driver/prepare.sh

```diff
@@ -142,16 +174,26 @@ start_services () {
        return
     fi
 
-    for l in $(echo "$ci_job_services" | jq -rc '.[]'); do
+    services=$(jq -rc '.services[]' "$JOB_RESPONSE_FILE")
+    job_vars=$(jq -r \
+        '.variables[] | select((.key | test("^(?!(CI|GITLAB)_)"))) | [.key, .value] | @sh' \
+        "$JOB_RESPONSE_FILE")
+
+    for l in $services; do
         # Using jq -er to fail of alias or name are not found
         alias_name=$(echo "$l" | jq -er '.alias | select(.)')
-        container_id="${container_id_base}-svc-${alias_name}"
         image_name=$(echo "$l" | jq -er '.name | select(.)')
-        # Using jq -r to allow entrypoint and command to be empty
+        container_id="${container_id_base}-svc-${alias_name}"
+
+        # Using jq -r to allow entrypoint, command, and variables to be empty
         container_entrypoint=$(echo "$l" | jq -r '.entrypoint | select(.)')
         container_command=$(echo "$l" | jq -r '.command | select(.)')
 
-        start_service "$alias_name" "$container_id" "$image_name" "$container_entrypoint" "$container_command"
+        # start_service will further process the variables, so just compact it
+        container_vars=$(echo "$l" | jq -r '.variables[] | [.key, .value] | @sh')
```

Comment:
[question] what does `@sh` do?

Reply:
It turns the `[.key, .value]` JSON array into space separated line that's a little easier to read into shell variables.

You can see what it looks like here: https://jqplay.org/s/GueUNFc73KTCKyc

Reply:
thanks! and thanks for sharing that site